lib/services/github.ts

```diff
@@ -324,6 +323,9 @@ export class GithubService extends WorkerClient<string> implements ServiceListen
 			});
 		};
 
+		if (!this.authToken) {
```

Comment:
Looks great, but can we just have a small comment here explaining this is on-demand auth.